cpp/src/gandiva/decimal_xlarge.cc

```diff
@@ -103,27 +103,33 @@ static BasicDecimal128 ConvertToDecimal128(int256_t in, bool* overflow) {
   return is_negative ? -result : result;
 }
 
+static int256_t GetScaleMultiplier(int scale) {
+  static constexpr int32_t kMaxScaleMultiplier = 2 * DecimalTypeUtil::kMaxPrecision;
+  DCHECK_GE(scale, 0);
+  DCHECK_LE(scale, kMaxScaleMultiplier);
+
+  // Compute the scale multipliers once.
+  static std::array<int256_t, kMaxScaleMultiplier + 1> multipliers =
+      ([]() -> std::array<int256_t, kMaxScaleMultiplier + 1> {
+        std::array<int256_t, kMaxScaleMultiplier + 1> values;
+        values[0] = 1;
+        for (auto idx = 1; idx <= kMaxScaleMultiplier; idx++) {
```

Comment:
Don't use auto on primitive types.

Reply:
curious, why the restriction ? is this from the coding standard or for efficiency ?

Reply:
fixed

Reply:
Mostly for readability (and future refactoring), and making sure there's no signed/unsigned issues.